aspnet/dnx/projects.rst

```diff
@@ -1,8 +1,652 @@
+	
+In the ``commands`` section shown above, the EF assembly doesn't require any extra arguments so it is just the name of the assembly.
+
+The assembly listed in the commands should be pulled in by a package that your application depends on.	
+
+For more information on using and creating commands see :doc:`/dnx/commands`.
+
+Running your app
+----------------
+
+By specifying a command as an argument that is passed to the DNX, you can run your app locally. For instance, you can use the ``kestrel`` command (specified in the ``commands`` section of your project's *project.json* file) to run a Web Application as follows::
+
+    dnx . kestrel
+
+To view the running web application, you can open a browser and navigate to the path specified in the *project.json* file::
+
+	http://localhost:5001
+	
+To run a Console Application using the Command Window from the project directory, you can use the following statement::
+
+	dnx . run
+	
+To run a Web API application, you can use the following command from the Command Window::
+
+	dnx . web
+	
+In the case of the *web* command, the ``Microsoft.AspNet.Hosting`` assembly has a ``Main`` entry point method that is called whenever you run the web command. The rest of the values in the *web* command are used by the hosting assembly to host your web application.
+
+To specify the application base directory path, you can use the ``appbase`` option. For example::
+
+	dnx --appbase c:\vscodeprojects\myapp
+	
+For a Console Application, the entry point is the ``Main`` method of the ``Program`` class. For more information about the Web Application startup process, see :doc:`/fundamentals/startup`.
+
+
+When your run your app, you can specify whether to compile in Debug mode or in Release mode. You can specify the ``configuration`` option as follows::
+
+	dnx --configuration Debug
+
+For more compilation related information, see :ref:`dnx-projects-CompiliationSettings`).
+
+Building
+--------
+
+You use DNX projects to build NuGet packages. You can use the .NET Development Utility (DNU) to build, package, and publish DNX projects. 
+
+DNU build
+^^^^^^^^^
+
+The *project.json* file defines your package metadata, your project dependencies, and which frameworks that you want to target your build. 
+
+All the files in the folder are by default part of the project unless explicitly excluded in *project.json*. You specify which frameworks to target by using the "frameworks" property. The DNX will cross-compile for each specified framework and create the corresponding *lib* folder in the built NuGet package.
+
+Building a project produces the binary outputs for the project. 
+
+.. _dnx-projects-CompiliationSettings:
+
+Compilation settings
+^^^^^^^^^^^^^^^^^^^^^
+
+Compilation settings allow you to pass options through to the `.NET Compiler Platform <https://github.com/dotnet/roslyn>`_. The language version can be set in this section of the *project.json* file, as well as defines and other options.
```

Comment:
Suggestion: "Compilation settings allow you to pass options through to the compiler"

It may not always be Roslyn, it could be F# or some other community-provided compiler.